src/test/java/org/pesmypetcare/webservice/dao/FreqTrainingDaoTest.java

```diff
@@ -0,0 +1,381 @@
+
+            freqTrainingDao.getAllFreqTraining(owner, petName);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
+    public void shouldReturnAllFreqTrainingsBetweenDatesOnDatabaseWhenRequested() throws DatabaseAccessException,
+        ExecutionException, InterruptedException {
+        given(db.collection(anyString())).willReturn(usersRef);
+        given(usersRef.document(anyString())).willReturn(ownerRef);
+        given(ownerRef.collection(anyString())).willReturn(petsRef);
+        given(petsRef.document(anyString())).willReturn(petRef);
+        given(petRef.collection(anyString())).willReturn(freqTrainingsRef);
+        given(freqTrainingsRef.get()).willReturn(futureQuery);
+        given(futureQuery.get()).willReturn(querySnapshot);
+        given(querySnapshot.getDocuments()).willReturn(freqTrainingsDocuments);
+        given(freqTrainingsDocuments.iterator()).willReturn(it);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(false);
+
+        List<Map<String, Object>> list = freqTrainingDao.getAllFreqTrainingsBetween(owner, petName, date, date2);
+
+        assertEquals(freqTrainingList, list, "Should return a List containing all meals between two dates");
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetFreqTrainingsBetweenDatabaseReceivesInterruptedException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(freqTrainingsRef);
+            given(freqTrainingsRef.get()).willReturn(futureQuery);
+            willThrow(InterruptedException.class).given(futureQuery).get();
+
+            freqTrainingDao.getAllFreqTrainingsBetween(owner, petName, date, date2);
+        }, INTERRUPTED_EXC_MSG);
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetFreqTrainingsBetweenFromDatabaseReceivesExecutionException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(freqTrainingsRef);
+            given(freqTrainingsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            freqTrainingDao.getAllFreqTrainingsBetween(owner, petName, date, date2);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)